picard/formats/id3.py

```diff
@@ -622,11 +631,129 @@ def _save_tags(self, tags, filename):
         if config.setting['write_id3v23']:
             tags.update_to_v23()
             separator = config.setting['id3v23_join_with']
-            tags.save(filename, v2_version=3, v1=v1, v23_sep=separator)
+            try:
+                tags.save(filename, v2_version=3, v1=v1, v23_sep=separator)
+            except ValueError:
+                for frame_id in list(tags.keys()):
+                    if not (frame_id.isalnum() and frame_id.isupper()):
+                        log.warning("Removing invalid frame ID: %r", frame_id)
+                        del tags[frame_id]
+                tags.save(filename, v2_version=3, v1=v1, v23_sep=separator)
         else:
             tags.update_to_v24()
             tags.save(filename, v2_version=4, v1=v1)
 
+    def _sanitize_id3_frames(self, tags):
+        """This method attempts to fix various issues with ID3 frames:
+        1. Handle invalid LINK frames (convert to WXXX if possible)
```

Comment:
Perhaps I don't understand exactly what cases this is supposed to fix - so it might be helpful if it was spelled out in the comments here. 

Perhaps the most obvious confusion is that this PR talks specifically about issues with the LINK tag (see section 4.20 of the [ID3 2.4 spec](https://id3.org/id3v2.4.0-frames) but appears to be looking at all tags and not just LINK tags (and not even just tags starting with W (see section 4.3 of the [ID3 2.4 spec](https://id3.org/id3v2.4.0-frames) which might potentially have this issue). 

I would like to see the specific tags to be sanitized listed here and the specific issues and fixes spelled out.

Reply:
Yes, I agree. I'd also prefer if this would fix the specific case we have.

Also something is strange here: For some reason this issue seems only to happen when saving ID3 v2.3, but not witth v2.4 (at least if what I wrote on the ticket is still true). Looking at the specs for "Linked information" in  both ID3 v2.3 and v2.4 I don't really understand why it should be. The specs are very similar. According to the problem description (files containing a LINK frame without frame ID, only the null byte) the LINK frame would be invalid for both.

So maybe mutagen does something different here, but for some reason can successfully save it to ID3 v2.4.